<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="4.5940" tests="10" failures="7">
  <testsuite name="Root Suite" timestamp="2021-12-09T01:34:56" tests="0" time="0.0000" failures="0">
  </testsuite>
  <testsuite name="express_authentication" timestamp="2021-12-09T01:34:56" tests="7" file="C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\index.spec.js" time="0.0780" failures="4">
    <testcase name="express_authentication Should generate an API key and add it to file" time="0.0020" classname="Should generate an API key and add it to file">
      <failure message="Cannot read property &apos;status&apos; of undefined" type="TypeError"><![CDATA[TypeError: Cannot read property 'status' of undefined
    at module.exports (key-store.js:13:16)
    at Context.<anonymous> (test\index.spec.js:38:9)
    at processImmediate (internal/timers.js:464:21)]]></failure>
    </testcase>
    <testcase name="express_authentication Should generate 5 unique API Keys" time="0.0010" classname="Should generate 5 unique API Keys">
      <failure message="Cannot read property &apos;status&apos; of undefined" type="TypeError"><![CDATA[TypeError: Cannot read property 'status' of undefined
    at module.exports (key-store.js:13:16)
    at Context.<anonymous> (test\index.spec.js:51:13)
    at processImmediate (internal/timers.js:464:21)]]></failure>
    </testcase>
    <testcase name="express_authentication Should return the API Key back in response" time="0.0300" classname="Should return the API Key back in response">
      <failure message="expected &apos;0JgkVQqCj&apos; to equal &apos;AE9MIR8lwZ&apos;" type="AssertionError"><![CDATA[AssertionError: expected '0JgkVQqCj' to equal 'AE9MIR8lwZ'
    at C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\index.spec.js:74:32]]></failure>
    </testcase>
    <testcase name="express_authentication Should send a 401 if the x-api-key header is not present" time="0.0070" classname="Should send a 401 if the x-api-key header is not present">
      <failure message="Cannot read property &apos;callCount&apos; of undefined" type="TypeError"><![CDATA[TypeError: Cannot read property 'callCount' of undefined
    at C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\index.spec.js:84:22
    at processTicksAndRejections (internal/process/task_queues.js:95:5)]]></failure>
    </testcase>
    <testcase name="express_authentication Should send a 401 if the x-api-key is invalid" time="0.0050" classname="Should send a 401 if the x-api-key is invalid">
    </testcase>
    <testcase name="express_authentication Should validate header for protected routes" time="0.0170" classname="Should validate header for protected routes">
    </testcase>
    <testcase name="express_authentication Should not require auth headers for unprotected routes" time="0.0160" classname="Should not require auth headers for unprotected routes">
    </testcase>
  </testsuite>
  <testsuite name="payment check" timestamp="2021-12-09T01:34:56" tests="3" file="C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\payment.spec.js" time="4.5160" failures="3">
    <testcase name="payment check Should generate a random price" time="0.5070" classname="Should generate a random price">
      <failure message="expected 0 to deeply equal 1" type="AssertionError"><![CDATA[AssertionError: expected 0 to deeply equal 1
    at C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\payment.spec.js:43:40]]></failure>
    </testcase>
    <testcase name="payment check Should generate 5 random prices" time="2.0020" classname="Should generate 5 random prices">
      <failure message="Timeout of 2000ms exceeded. For async tests and hooks, ensure &quot;done()&quot; is called; if returning a Promise, ensure it resolves. (C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\payment.spec.js)" type="Error"><![CDATA[Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\payment.spec.js)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)]]></failure>
    </testcase>
    <testcase name="payment check Should return 5 promo codes" time="2.0070" classname="Should return 5 promo codes">
      <failure message="Timeout of 2000ms exceeded. For async tests and hooks, ensure &quot;done()&quot; is called; if returning a Promise, ensure it resolves. (C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\payment.spec.js)" type="Error"><![CDATA[Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\naire\OneDrive\Escritorio\DADSTA\Proyecto\proyect\test\payment.spec.js)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)]]></failure>
    </testcase>
  </testsuite>
</testsuites>